crates/core/src/db/relational_db.rs

```diff
@@ -570,69 +564,70 @@ impl RelationalDB {
     /// [`Self::commit_tx`], otherwise the database will be left in an invalid
     /// state. See also [`Self::with_auto_commit`].
     #[tracing::instrument(skip_all)]
-    pub fn begin_mut_tx(&self, isolation_level: IsolationLevel) -> MutTx {
+    pub fn begin_mut_tx(&self, isolation_level: IsolationLevel, workload: Workload) -> MutTx {
         log::trace!("BEGIN MUT TX");
-        let r = self.inner.begin_mut_tx(isolation_level);
+        let r = self.inner.begin_mut_tx(isolation_level, workload);
         log::trace!("ACQUIRED MUT TX");
         r
     }
 
     #[tracing::instrument(skip_all)]
-    pub fn begin_tx(&self) -> Tx {
+    pub fn begin_tx(&self, workload: Workload) -> Tx {
         log::trace!("BEGIN TX");
-        let r = self.inner.begin_tx();
+        let r = self.inner.begin_tx(workload);
         log::trace!("ACQUIRED TX");
         r
     }
 
     #[tracing::instrument(skip_all)]
-    pub fn rollback_mut_tx(&self, ctx: &ExecutionContext, tx: MutTx) {
+    pub fn rollback_mut_tx(&self, tx: MutTx) {
         log::trace!("ROLLBACK MUT TX");
-        self.inner.rollback_mut_tx(ctx, tx)
+        self.inner.rollback_mut_tx(tx)
     }
 
     #[tracing::instrument(skip_all)]
-    pub fn rollback_mut_tx_downgrade(&self, ctx: &ExecutionContext, tx: MutTx) -> Tx {
+    pub fn rollback_mut_tx_downgrade(&self, tx: MutTx, workload: Workload) -> Tx {
         log::trace!("ROLLBACK MUT TX");
-        self.inner.rollback_mut_tx_downgrade(ctx, tx)
+        self.inner.rollback_mut_tx_downgrade(tx, workload)
     }
 
     #[tracing::instrument(skip_all)]
-    pub fn release_tx(&self, ctx: &ExecutionContext, tx: Tx) {
+    pub fn release_tx(&self, tx: Tx) {
         log::trace!("RELEASE TX");
-        self.inner.release_tx(ctx, tx)
+        self.inner.release_tx(tx)
     }
 
     #[tracing::instrument(skip_all)]
-    pub fn commit_tx(&self, ctx: &ExecutionContext, tx: MutTx) -> Result<Option<TxData>, DBError> {
+    pub fn commit_tx(&self, tx: MutTx) -> Result<Option<TxData>, DBError> {
         log::trace!("COMMIT MUT TX");
 
         // TODO: Never returns `None` -- should it?
-        let Some(tx_data) = self.inner.commit_mut_tx(ctx, tx)? else {
+        let ctx = tx.ctx.clone();
```

Comment:
Technically `do_durability` only needs a `ReducerContext`. Could you update its function signature? That way we only need to clone the `ReducerContext` here instead of the `ExecutionContext`.

Reply:
Changed.